File: src/eth/ws_eth_findTokens/Worker.cs
using eth_shared;

namespace ws_eth_findTokens
{
    public class Worker : BackgroundService
    {
        private readonly ILogger _logger;
        private readonly IServiceScopeFactory serviceScopeFactory;

        public Worker(
            ILogger<Worker> logger,
            IServiceScopeFactory serviceScopeFactory
            )
        {
            _logger = logger;
            this.serviceScopeFactory = serviceScopeFactory;
        }

        protected override async Task ExecuteAsync(CancellationToken stoppingToken)
        {
            using (IServiceScope scope = serviceScopeFactory.CreateScope())
            {
                IScopedProcessingService scopedProcessingService =
                    scope.
                    ServiceProvider.
                    GetRequiredKeyedService<IScopedProcessingService>("WorkerScoped");

                await scopedProcessingService.DoWorkAsync(stoppingToken);
            }
        }
    }
}

File: src/f#/eth/wsSwaps/Worker.fs
namespace wsSwaps

open System.Threading
open System.Threading.Tasks
open System.Collections.Generic

open Microsoft.Extensions.Hosting
open Microsoft.Extensions.Logging
open Microsoft.Extensions.DependencyInjection

open IScopedProcessingService

type Worker(
        logger: ILogger<Worker>, 
        serviceScopeFactory: IServiceScopeFactory) =
    inherit BackgroundService()

    override this.ExecuteAsync(stoppingToken: CancellationToken) =
        task {
            while not stoppingToken.IsCancellationRequested do
                use scope = serviceScopeFactory.CreateScope()
                let serviceFactory = scope.ServiceProvider.GetRequiredService<IDictionary<string, IScopedProcessingService>>()
                let scopedProcessingService = serviceFactory.["WorkerService1"]
                do! scopedProcessingService.DoWorkAsync(stoppingToken)
                do! Task.Delay(1000, stoppingToken)
        }